docs/blog/2019-03-27-security-for-modern-web-frameworks/index.md

```diff
@@ -0,0 +1,135 @@
+
+Why address this scenario? Two reasons:
+
+- **Simplicity** - Consider a small blog or non-profit site where authentication is not already needed. To develop it would require a lot of time and long term investment. Users authenticating to a small blog just to hit "like" is a poor user experience and probably a hurtle they will not bother jumping.
+
+- **Privacy** - No matter the size of your platform, requiring users to authenticate means having to secure private data: this requires time and money. Alternatively, as a platform, you may decide your users' privacy is more important then better security controls.
+
+Using Medium's "clap" feature as example, what are the security issues? First, we would need some kind of database to store the claps. Do we leave no security mechanism on this database for reads and writes? We could add security, but then the necessary API key has to be shipped in the code and the user can access it. Either way, your more nefarious users could access this database; reading and (more importantly) writing to their devious heart's content.
+
+There is a small blog that does this very thing. They build the Firebase credentials into the code at build time. Because this is a small blog, and there is no real incentive for someone to add or delete claps, this blogger is probably safe. What is the worse case scenario? Someone deletes their claps? Gives all their posts a million claps? Maybe this is defeating the point of a security post. But security is about threat assessment. This blogger is clearly not worried about a nation state hacking their claps or someone gaming the system. Nothing but authentication is going to solve those kinds of issues
+
+Assuming a nation state is not trying to attack you, what are you options for better security? You need an API.
+
+One great option is to check out Netlify. They will provide great static hosting and allow you to deploy serverless functions right along side your Gatsby site. They are not the only ones: Azure, Google, Amazon, Zeit.co, and others all have API solutions.
+
+What does this API allow? First, you can secure your database. Any API keys can be secured on your serverless...server 😭. The point is, they will be secure. But what does this gain you? After all, instead of pillaging your database directly you have given them an API through which to pillage.
+
+**Control:** You have gained the ability to add rate limiting, better logging, auto blocking of IPs, etc. You can throw the proverbial "secure API handbook" (except for the authentication part) at your users to make sure they behave. More aggressively, you could use a secure cookie for your site to uniquely and anonymously identify each user. This would assist in the previously mentioned security practices and, while not hard for a user to delete a cookie, it adds another layer of security.
```

Comment:
1) is there a link to a “Secure API Handbook”? I could not find a matching on google
2) i dont understand this sentence :( why i should throw a “handbook” at users? They need to read them? I expect more “at users login/authentification implementation”

Reply:
@muescha Okay, this is hilarious! It must just be an american idiom. to "Throw the book" at someone or something means to use everything in it. It came from the court system, [see dictionary.com](https://www.dictionary.com/browse/throw-the-book-at-someone).

the handbook doesn't exist...that's why it's "proverbial", meaning from a proverb, or "not real".

@marcysutton Thoughts on keeping or eliminating these american idioms?

Reply:
@moonmeister if there's a way to write it that doesn't depend on slang or idioms, it would definitely make the text more inclusive!

Reply:
@marcysutton Yeah, I'll re-write it. Maybe see if I can find a good resource on API security even.

Reply:
There is this cheat-sheet we could link to: https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/REST_Security_Cheat_Sheet.md